frontend/result.go

```diff
@@ -2,12 +2,36 @@ package frontend
 
 import (
 	"github.com/moby/buildkit/solver"
+	digest "github.com/opencontainers/go-digest"
 )
 
+type Attestation struct {
+	PredicateType string
+	PredicateRef  solver.ResultProxy
+	PredicatePath string
+	Subjects      []AttestationSubject
+}
+
+type AttestationSubject interface {
+	isAttestationSubject()
+}
+
+type AttestationSubjectRaw struct {
+	Name   string
+	Digest digest.Digest
+}
+
+func (as *AttestationSubjectRaw) isAttestationSubject() {}
+
+type AttestationSubjectSelf struct{}
```

Comment:
Why are these defined twice? Move to a separate package?

I think `AttestationSubjectRaw` can be linked from https://github.com/in-toto/in-toto-golang/blob/master/in_toto/model.go#L966

Reply:
Ah, so we should also probably use the `slsa.DigestSet` instead of a singular `digest.Digest` throughout, and down to the protobuf layer.

Reply:
It is a very simple type, I am not sure if it's worse adding a new dependency just for that type here. This PR already introduces intoto types as a dependency, which does pull in slsa types, so it's not going to be a new one for sure.
In my experience there is not a lot of value in importing intoto types Vs redefining locally, as they are really quite minimal. I don't think we need to worry about this now, mostly just wanted to mention this as an observation.

Reply:
Hm `DigestSet` as a structure is a fiddly to work with.
I think we might actually be better storing `[]digest.Digest` to fit in with the rest of the protobuf protocol, and then translating properly to a map in the exporter.